plugins/obs-filters/data/blend_add_filter.effect

```diff
@@ -37,7 +38,7 @@ float4 PSAddImageRGBA(VertDataOut v_in) : TARGET
 	float4 rgba = image.Sample(textureSampler, v_in.uv) * color;
 
 	float4 targetRGB = target.Sample(textureSampler, v_in.uv2);
-	rgba.rgb = saturate(rgba.rgb + targetRGB.rgb);
+	rgba.rgb = lerp(rgba.rgb, saturate(rgba.rgb + targetRGB.rgb), opacity * 0.01);
```

Comment:
Scaling of opacity should be done on the CPU, not GPU.

Reply:
> > In my review `color |= (uint32_t)(((double)opacity) * 2.55) << 24;` says if no color, use opacity and create white for color ( color = opacity * 255,255,255). This is for a default value.
> 
> It's actually more like color = float4(color.rgb, color.a | opacity). Maybe the initial implementation could assume the high alpha bits were 0 because bitwise or doesn't make sense otherwise.
> 
> The Opacity setting in current OBS does something. If you load an image into the filter, don't touch the color picker, and change the Opacity setting, the output image becomes affected.

As buggy as that is, it is because there is no color so it is created by Opacity. 

That is why I say that it really evaluates to  "if there is no color, then create a default color from Opacity."

However, that is not in any sense Opacity it is a color. It just happens to have a default Opacity of 1, and the bitwise "<<24"  pushes that to a float4 white.